src/Command/Cache/DoctrineResultCacheWarmerCommand.php

```diff
@@ -119,47 +176,85 @@ protected function execute(InputInterface $input, OutputInterface $output)
-        } while ($offset <= $numberOfTotalUsers + $batchSize);
+            $offset += $this->batchSize;
+        } while ($offset <= $numberOfTotalUsers + $this->batchSize);
 
-        $style->success(
+        $this->symfonyStyle->success(
             sprintf(
                 '%s result cache entries have been successfully warmed up.',
                 $numberOfWarmedUpCacheEntries
             )
         );
 
-        $style->note(sprintf('Took: %s', $this->stopWatch(self::NAME)));
+        $this->symfonyStyle->note(sprintf('Took: %s', $this->stopWatch(self::NAME)));
 
         return 0;
     }
 
-    private function getFindAllUsersNameQuery(int $offset, int $batchSize): Query
+    private function getFindAllUsernameQuery(int $offset): Query
     {
-        return $this->entityManager
+        $queryBuilder = $this->entityManager
             ->createQueryBuilder()
             ->select('u.username')
-            ->from(User::class, 'u')
+            ->from(User::class, 'u');
+
+        if (!empty($this->userIds)) {
+            $queryBuilder
+                ->where('u.id IN (:userIds)')
+                ->setParameter('userIds', $this->userIds);
+        }
+
+        if ($this->lineItemIds) {
+            $queryBuilder
+                ->distinct()
+                ->leftJoin('u.assignments', 'a')
+                ->leftJoin('a.lineItem', 'l')
+                ->leftJoin('l.infrastructure', 'i')
+                ->where('l.id IN (:lineItemIds)')
+                ->setParameter('lineItemIds', $this->lineItemIds);
+        }
+
+        return $queryBuilder
             ->setFirstResult($offset)
-            ->setMaxResults($batchSize)
+            ->setMaxResults($this->batchSize)
             ->getQuery()
             ->setHydrationMode(Query::HYDRATE_SINGLE_SCALAR);
     }
 
-    private function getTotalNumberOfUsers(): int
+    private function getNumberOfTotalUsers(): int
     {
-        return (int)$this->entityManager
-            ->createQueryBuilder()
-            ->select('COUNT(u.id)')
-            ->from(User::class, 'u')
+        $queryBuilder =
+            $this->entityManager
+                ->createQueryBuilder()
+                ->select('COUNT(u.id) AS number_of_users')
+                ->from(User::class, 'u');
+
+        if ($this->userIds) {
+            $queryBuilder
+                ->where('u.id IN (:userIds)')
+                ->setParameter('userIds', $this->userIds);
+        }
+
+        if ($this->lineItemIds) {
+            $queryBuilder
+                ->leftJoin('u.assignments', 'a')
+                ->leftJoin('a.lineItem', 'l')
+                ->leftJoin('l.infrastructure', 'i')
+                ->where('l.id IN (:lineItemIds)')
+                ->setParameter('lineItemIds', $this->lineItemIds);
+        }
+
+        $result = $queryBuilder
             ->getQuery()
-            ->getSingleScalarResult();
+            ->getOneOrNullResult();
+
+        return (int)$result['number_of_users'];
```

Comment:
getOneOrNullResult may return null, in this case `$result['number_of_users']` seems unsafe.

Reply:
Indeed, it will throw a notice, if null is returned.